SCClassLibrary/Common/GUI/Base/Grid.sc

```diff
@@ -290,7 +300,18 @@ GridLines {
 	var <>spec;
 
 	*new { arg spec;
-		^super.newCopyArgs(spec.asSpec)
+		spec = spec.asSpec;
+		spec.warp.switch(
+			LinearWarp(), {
+				^super.newCopyArgs(spec)
+			},
+			ExponentialWarp(), {
+				^ExponentialGridLines(spec)
+			},
+			{
+				^super.newCopyArgs(spec)
+			} // all other Warps
+		)
```

Comment:
The most canonical way would probably be to implement a method:

```
Warp {
     gridClass { ^GridLines }
}
ExponentialWarp {
     gridClass { ^ExponentialGridLines }
}
}
```

then you can just call:

```
^spec.warp.gridClass.new(this)
```
(maybe catching nil)

Reply:
How does it look now?